### PR TITLE
Database connection and dummy init

### PR DESCRIPTION
**Pull request by asntr**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,6 +35,7 @@
     "eslint-plugin-node": "^5.1.0",
     "eslint-plugin-promise": "^3.5.0",
     "eslint-plugin-standard": "^3.0.1",
+    "mongoose": "^4.11.1",
```
Comment (Drapegnik) [inline package.json]: duplication, please remove this

File: src/db/db.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import mongoose from 'mongoose';
+
+import config from '../config';
+
+export default function () {
+  mongoose.connect(config.mongodb.url, {
+    useMongoClient: true
+  });
+
+  mongoose.connection.on('connected', () => {
```
Comment (Drapegnik) [inline src/db/db.js]: you can omit curly brackets here

File: src/db/db.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    console.log(`Mongoose: connected to ${config.mongodb.url}`);
+  });
+
+  mongoose.connection.on('error', (err) => {
+    console.log(`Mongoose: connection error: ${err}`);
+  });
+
+  mongoose.connection.on('disconnected', () => {
```
Comment (Drapegnik) [inline src/db/db.js]: same as above

File: init-db.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable no-console */
+
+import * as db from 'src/db';
+
+Promise.resolve(db.connectDb())
+  .then(mongoose => db.dropCollections(mongoose))
+  .then((collections) => {
```
Comment (Drapegnik) [inline init-db.js]: use simple arrow function syntax: `bar => foo(bar)`

File: src/db/db.js (hunk 3 above)
Comment (Drapegnik) [inline src/db/db.js]: check: http://eslint.org/docs/rules/arrow-parens.html

**Comment by Drapegnik** [comment]
do you run code?
```
Drapegnik@MacBook-Pro-Ivan:~/projects/babajka/backend (database-init *)$ node_modules/.bin/babel-node init-db.js
module.js:487
    throw err;
    ^

Error: Cannot find module 'src/db'
```

looks like you should write './src/db' or setup babel plugin for global exports (it's better solution)

**Comment by Drapegnik** [comment]
screen shot 2017-07-09 at 02 56 32

```
Drapegnik@MacBook-Pro-Ivan:~/projects/babajka/backend (database-init *)$ node_modules/.bin/babel-node init-db.js
Mongoose: drop 0 collections: 
```

**Comment by Drapegnik** [comment]
it'll be cool if you add some test cases, for proof that everything is work, like:
* init-db drop all collections, and create new one with some data
* server connect to db and select some data from this new collection

just for sure that all is okay

**Comment by Drapegnik** [comment]
add `init-db.js` to `lint` task, and update some rules, please follow it

File: package.json [hunk 5]
```diff
@@ -29,11 +30,9 @@
...
     "babel-preset-stage-2": "^6.24.1",
     "eslint": "^4.1.1",
-    "eslint-config-standard": "^10.2.1",
-    "eslint-plugin-import": "^2.6.1",
-    "eslint-plugin-node": "^5.1.0",
-    "eslint-plugin-promise": "^3.5.0",
-    "eslint-plugin-standard": "^3.0.1",
+    "eslint-config-airbnb": "^15.0.2",
+    "eslint-config-airbnb-base": "^11.2.0",
+    "mongoose": "^4.11.1",
```
Comment (Drapegnik) [inline package.json]: duplication, please remove it

**Comment by Drapegnik** [comment]
here is a babel plugin for cool import paths https://www.npmjs.com/package/babel-plugin-global-require

File: src/db/db.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable no-console */
+
+import mongoose from 'mongoose';
+
+import config from '../config';
+
+export default function () {
+  mongoose.connect(config.mongodb.url, {
+    useMongoClient: true
```
Comment (Drapegnik) [inline src/db/db.js]: please extract it in `config/config.json`

File: src/db/db.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+  mongoose.connection.on('error', (err) => {
+    console.log(`Mongoose: connection error: ${err}`);
+  });
+
+  mongoose.connection.on('disconnected', () => {
+    console.log('Mongoose: connection disconnected');
+  });
+
+  return mongoose;
```
Comment (Drapegnik) [inline src/db/db.js]: i think better rename this file to `mongoose`, cause you export from it exactly `mongoose` object, but you can argue

File: src/config/config.json [hunk 8]
```diff
@@ -1,4 +1,10 @@
 {
   "port": 3000,
-  "publicPath": "public"
+  "publicPath": "public",
+  "mongodb": {
+    "url": "mongodb://localhost/babajka",
+    "options": {
+      "useMongoClient": true
```
Comment (Drapegnik) [inline src/config/config.json]: what do this option?

File: .babelrc [hunk 9]
```diff
@@ -2,5 +2,10 @@
   "presets": [
     "es2015",
     "stage-2"
+  ],
+  "plugins": [
+    ["global-require", {
```
Comment (Drapegnik) [inline .babelrc]: Are you use it now?

**Comment by Drapegnik** [comment]
Updated, @asntr please check it.

**Comment by asntr** [comment]
Wow! Almost everything looks pretty cool, but I feel uncomfortable with tests in `src` directory. @UladBohdan, leave your opinion, please.

**Comment by Drapegnik** [comment]
imho, if we use feature based structure -  it's convinion have User folder, and ALL about user into it (eg model, controller, routes and tests)

**Comment by asntr** [comment]
I see. But tests do not provide any functionality of modules that they test. Moreover, in future, some tests can consist of interaction cases between different modules. I see tests as a separated feature. Anyway, you don't put `init-db.js` into `src/db`, do you?

**Comment by Drapegnik** [comment]
[connect database](https://trello.com/c/JA3ExQoX/7-connect-database)

**Comment by UladBohdan** [comment]
I don't know, what's the common approach to testing in JS.
But generally if we're talking about unit testing it's better to have the code and the test together and not to create two hierarchies (one with code, another with tests). So unit tests I would keep together with the code, probably with the same filename (with 'test' modifier). These tests must only test these files.
@asntr for more complicated tests (to test general functionality) we will probably have `tests` folder then.

And it seems weird to me to have `init-db.js` file in the root of the repository. Why?

**Comment by UladBohdan** [comment]
Another thing to consider is that probably that's time to gitignore `config.json` file. Configuration must never be in the repo, although that's not yet a problem (no production config).

The most up-to-date `config.json` may be stored in slack, for example.

**Pull request merged**